File: tests/python/pages/productpage.py
from playwright.sync_api import Page
from tests.python.config import version

class Product():

    def __init__(self, page: Page):
        self.page = page

    def navigate(self, product_id: int):
        self.page.goto(f'https://{version}.practicesoftwaretesting.com/#/product/{product_id}')

File: tests/typescript/pages/product-page.ts
import type { Locator, Page } from "@playwright/test";
import { baseURL } from "../config";

export class ProductPage {
    readonly page: Page
    readonly productName: Locator

    constructor(page: Page) {
        this.page = page
        this.productName = page.getByRole('heading').and(page.getByTestId('product-name'))
    }

}